src/help/utils/apiSnippets.tsx

```diff
@@ -0,0 +1,307 @@
+    isJSON: headers.get('Accept') === 'application/json',
+  };
+};
+
+type SnippetGeneratorPlugin = {
+  fn: Partial<
+    Record<
+      `requestSnippetGenerator_${keyof typeof requestSnippets.generators}`,
+      (request: RequestMap) => string
+    >
+  >;
+};
+
+// PYTHON
+const SnippetGeneratorPythonPlugin: SnippetGeneratorPlugin = {
+  fn: {
+    // use `requestSnippetGenerator_` + key from config (node_native) for generator fn
+    requestSnippetGenerator_python_requests: (request) => {
+      const { url, searchParams, method, headers, isJSON } =
+        requestExtract(request);
+
+      if (method === 'POST') {
+        return '# No automatic snippet, work in progress';
+      }
+
+      return `import requests, sys${isJSON ? ', json' : ''}
+${
+  searchParams
+    ? `
+params = ${JSON.stringify(
+        Object.fromEntries(searchParams),
+        (key, value) => {
+          if (key === 'fields') {
+            return value.split(',');
+          }
+          return value;
+        },
+        2
+      )}`
+    : ''
+}
+headers = ${JSON.stringify(Object.fromEntries(headers.entries()), null, 2)}
+base_url = "${url.origin + url.pathname}"
+
+response = requests.get(base_url, headers=headers, params=params)
+if not r.ok:
+  r.raise_for_status()
+  sys.exit()
+
+data = response.${isJSON ? 'json()' : 'text'}
+
+print(${isJSON ? 'json.dumps(data, indent=2)' : 'data'})
+`;
+    },
+  },
+};
+
+// JAVASCRIPT
+const SnippetGeneratorNodeJsPlugin: SnippetGeneratorPlugin = {
+  fn: {
+    // use `requestSnippetGenerator_` + key from config (node_native) for generator fn
+    requestSnippetGenerator_js_fetch: (request) => {
+      const { url, searchParams, method, headers, body, isJSON } =
+        requestExtract(request);
+
+      if (method === 'POST') {
+        return '// No automatic snippet, work in progress';
+      }
+
+      return `// requirement: NodeJS >= 14.8.0, or Firefox, Chrome, or Edge >= 89
+${
+  searchParams
+    ? `
+const searchParams = new URLSearchParams(${JSON.stringify(
+        Object.fromEntries(searchParams),
+        (key, value) => {
+          if (key === 'fields') {
+            return value.split(',');
+          }
+          return value;
+        },
+        2
+      )});`
+    : ''
+}
+const url = \`${url.origin + url.pathname}${searchParams ? `?$\{searchParams}` : ''}\`;
+
+const response = await fetch(url, {
+  headers: ${indent(JSON.stringify(Object.fromEntries(headers.entries()), null, 2))},${
+    method === 'GET'
+      ? ''
+      : `
+  method: "${method}",`
+  }${
+    body
+      ? `
+  body: new FormData(${indent(JSON.stringify(body, null, 2))}),`
+      : ''
+  }
+});
+if (!response.ok) {
+  throw new Error(\`$\{response.status}: $\{response.statusText}\`);
+}
+
+const data = await response.${isJSON ? 'json' : 'text'}();
+
+console.log(data);
+`;
+    },
+  },
+};
+
+// PERL
+const SnippetGeneratorPerlPlugin: SnippetGeneratorPlugin = {
```

Comment:
I had to install the following with cpanm to make this work
```
cpanm Type::Tiny JSON IO::Socket::SSL
```
I'm not sure if it's worth mentioning this to users or if it is implied. A comment for us devs in the future could be useful otherwise.

Reply:
I've added a comment at the top